### PR TITLE
end to end notebook (simplified, cleaned up taxi notebook) and NLP notebook

### PR DESCRIPTION
**Pull request by beckernick**

### CODE REVIEW

File: Notebooks/06-End_to_End.ipynb [hunk 1]
```diff
@@ -0,0 +1,699 @@
...
+   "source": [
+    "def haversine_dist(df):\n",
+    "    df['h_distance']= cuspatial.haversine_distance(\n",
+    "        df['pickup_longitude'],\n",
+    "        df['pickup_latitude'],\n",
+    "        df['dropoff_longitude'],\n",
+    "        df['dropoff_latitude']\n",
+    "    )\n",
+    "    df['h_distance']= df['h_distance'].astype('float32')\n",
+    "    return df\n",
```
Comment (VibhuJawa) [inline Notebooks/06-End_to_End.ipynb:374]: Can we replace this distance to a native cupy kernel?   

The reason is that this makes a CPU to GPU comparison easier and removes `cuspatial` as a dependency.  

```python
def cudf_haversine_distance(lon1, lat1, lon2, lat2):

    lon1, lat1, lon2, lat2 = map(cp.radians, [lon1, lat1, lon2, lat2])

    newlon = lon2 - lon1
    newlat = lat2 - lat1

    haver_formula = cp.sin(newlat/2.0)**2 + cp.cos(lat1) * cp.cos(lat2) * cp.sin(newlon/2.0)**2

    dist = 2 * cp.arcsin(cp.sqrt(haver_formula ))
    km = 6367 * dist #6367 for distance in KM for miles use 3958
    return km


def haversine_dist(df):
    df['h_distance']= cudf_haversine_distance(
        df['pickup_longitude'],
        df['pickup_latitude'],
        df['dropoff_longitude'],
        df['dropoff_latitude']
    )
    df['h_distance']= df['h_distance'].astype('float32')
    return df
```


I can confirm that the run time and accuracy both remain the same with this function.

Comment (beckernick) [reply Notebooks/06-End_to_End.ipynb:374]: We can, but I think there is value in highlighting that cuSpatial exists. Perhaps in your comparison notebook you could do this but in this one we keep it like this?

Comment (VibhuJawa) [reply Notebooks/06-End_to_End.ipynb:374]: That works too.

Comment (beckernick) [reply Notebooks/06-End_to_End.ipynb:374]: Discussed offline. Keeping as is

**Pull request merged**